### ISSUE TITLE
Return details about TLS (possibly full cert chain)

### ISSUE THREAD

**Issue by hellais** [measurement-kit/measurement-kit#1493]
Measurement Kit should return TLS information when it does the TLS handshake.

Ideally it would return the full cert chain.

**Comment by hellais** [comment]
Related: https://github.com/measurement-kit/measurement-kit/issues/1420

**Comment by bassosimone** [comment]
Of course, by hooking into the callbacks, one can get as much data as desired. It is true that there are other ways to extract the certificate using OpenSSL API. But this approach is as close as possible to "PCAPs or it didn't happen" without requiring root privileges, that I'd recommend using it during the SSL handshake to have low level data that we can use in case of issues.

This is an example of the data that I could extract with this approach, where the SSL packet with the certificate is quite obvious:

```
time: 1518616328.937972 operation: read
    0b 00 0f fb 00 0f f8 00  08 42 30 82 08 3e 30 82  |.........B0..>0.|
    07 26 a0 03 02 01 02 02  08 30 2a 6f 87 1d 13 15  |.&.......0*o....|
    28 30 0d 06 09 2a 86 48  86 f7 0d 01 01 0b 05 00  |(0...*.H........|
    30 49 31 0b 30 09 06 03  55 04 06 13 02 55 53 31  |0I1.0...U....US1|
    13 30 11 06 03 55 04 0a  13 0a 47 6f 6f 67 6c 65  |.0...U....Google|
    20 49 6e 63 31 25 30 23  06 03 55 04 03 13 1c 47  |.Inc1%0#..U....G|
    6f 6f 67 6c 65 20 49 6e  74 65 72 6e 65 74 20 41  |oogle.Internet.A|
    75 74 68 6f 72 69 74 79  20 47 32 30 1e 17 0d 31  |uthority.G20...1|
    38 30 31 33 30 30 38 35  35 35 30 5a 17 0d 31 38  |80130085550Z..18|
    30 34 32 34 30 38 33 30  30 30 5a 30 66 31 0b 30  |0424083000Z0f1.0|
    09 06 03 55 04 06 13 02  55 53 31 13 30 11 06 03  |...U....US1.0...|
    55 04 08 0c 0a 43 61 6c  69 66 6f 72 6e 69 61 31  |U....California1|
    16 30 14 06 03 55 04 07  0c 0d 4d 6f 75 6e 74 61  |.0...U....Mounta|
    69 6e 20 56 69 65 77 31  13 30 11 06 03 55 04 0a  |in.View1.0...U..|
    0c 0a 47 6f 6f 67 6c 65  20 49 6e 63 31 15 30 13  |..Google.Inc1.0.|
    06 03 55 04 03 0c 0c 2a  2e 67 6f 6f 67 6c 65 2e  |..U....*.google.|
    63 6f 6d 30 82 01 22 30  0d 06 09 2a 86 48 86 f7  |com0.."0...*.H..|
    0d 01 01 01 05 00 03 82  01 0f 00 30 82 01 0a 02  |...........0....|
    82 01 01 00 a4 aa 90 a8  c8 14 fb 82 de 9f 7f 32  |...............2|
    32 98 f1 6f 6f b4 79 ee  6b 40 0f d3 ed 5f 5d 15  |2..oo.y.k@..._].|
    19 fa b1 c3 25 f9 e0 eb  14 fd 08 0a c3 7c 13 76  |....%........|.v|
    bb 2a ec 7f 12 f3 af ae  f6 44 b4 f6 77 cb 02 e7  |.*.......D..w...|
    88 81 d9 94 76 ee 17 a6  2f 3c 23 15 16 11 7d b3  |....v.../<#...}.|
    c3 18 0f e4 1b 1b f3 03  13 55 25 0c b7 83 90 2c  |.........U%....,|
    d4 7e 92 08 f6 fb 2c 0f  ec f1 76 13 e0 2b 99 a1  |.~....,...v..+..|
    59 3d e8 89 31 41 c7 a1  e7 92 da 12 ee c3 40 56  |Y=..1A........@V|
    e2 59 e9 e1 eb 9a e8 0a  c2 c3 4d 9a c1 f4 51 57  |.Y........M...QW|
    30 c3 1b dc f7 d7 47 22  b3 51 f2 b5 70 2c 6f dc  |0.....G".Q..p,o.|
    20 85 e4 5b 04 75 ff 16  ce 38 ff 2d d5 c3 be 06  |...[.u...8.-....|
    51 15 2d c4 a4 f2 46 e2  6d bb 43 d3 d7 4c dc 64  |Q.-...F.m.C..L.d|
    3d f5 c6 e8 51 a9 59 e5  b5 b7 a3 7d a2 c8 8e fc  |=...Q.Y....}....|
    d1 46 e6 2c 15 a2 d7 8b  de 35 fe 15 37 ad ae 59  |.F.,.....5..7..Y|
    69 d9 86 3d fb 80 91 d0  54 6f 56 4a 7e 8b 02 7c  |i..=....ToVJ~..||
    12 d6 6d 38 ca f2 66 3a  18 66 28 91 a8 03 25 87  |..m8..f:.f(...%.|
    aa 1b 7e 1d 02 03 01 00  01 a3 82 05 0b 30 82 05  |..~..........0..|
    07 30 13 06 03 55 1d 25  04 0c 30 0a 06 08 2b 06  |.0...U.%..0...+.|
    01 05 05 07 03 01 30 82  03 e1 06 03 55 1d 11 04  |......0.....U...|
    82 03 d8 30 82 03 d4 82  0c 2a 2e 67 6f 6f 67 6c  |...0.....*.googl|
    65 2e 63 6f 6d 82 0d 2a  2e 61 6e 64 72 6f 69 64  |e.com..*.android|
    2e 63 6f 6d 82 16 2a 2e  61 70 70 65 6e 67 69 6e  |.com..*.appengin|
    65 2e 67 6f 6f 67 6c 65  2e 63 6f 6d 82 12 2a 2e  |e.google.com..*.|
    63 6c 6f 75 64 2e 67 6f  6f 67 6c 65 2e 63 6f 6d  |cloud.google.com|
    82 14 2a 2e 64 62 38 33  33 39 35 33 2e 67 6f 6f  |..*.db833953.goo|
    67 6c 65 2e 63 6e 82 06  2a 2e 67 2e 63 6f 82 0e  |gle.cn..*.g.co..|
    2a 2e 67 63 70 2e 67 76  74 32 2e 63 6f 6d 82 16  |*.gcp.gvt2.com..|
    2a 2e 67 6f 6f 67 6c 65  2d 61 6e 61 6c 79 74 69  |*.google-analyti|
    63 73 2e 63 6f 6d 82 0b  2a 2e 67 6f 6f 67 6c 65  |cs.com..*.google|
    2e 63 61 82 0b 2a 2e 67  6f 6f 67 6c 65 2e 63 6c  |.ca..*.google.cl|
    82 0e 2a 2e 67 6f 6f 67  6c 65 2e 63 6f 2e 69 6e  |..*.google.co.in|
    82 0e 2a 2e 67 6f 6f 67  6c 65 2e 63 6f 2e 6a 70  |..*.google.co.jp|
    82 0e 2a 2e 67 6f 6f 67  6c 65 2e 63 6f 2e 75 6b  |..*.google.co.uk|
    82 0f 2a 2e 67 6f 6f 67  6c 65 2e 63 6f 6d 2e 61  |..*.google.com.a|
    72 82 0f 2a 2e 67 6f 6f  67 6c 65 2e 63 6f 6d 2e  |r..*.google.com.|
    61 75 82 0f 2a 2e 67 6f  6f 67 6c 65 2e 63 6f 6d  |au..*.google.com|
    2e 62 72 82 0f 2a 2e 67  6f 6f 67 6c 65 2e 63 6f  |.br..*.google.co|
    6d 2e 63 6f 82 0f 2a 2e  67 6f 6f 67 6c 65 2e 63  |m.co..*.google.c|
    6f 6d 2e 6d 78 82 0f 2a  2e 67 6f 6f 67 6c 65 2e  |om.mx..*.google.|
    63 6f 6d 2e 74 72 82 0f  2a 2e 67 6f 6f 67 6c 65  |com.tr..*.google|
    2e 63 6f 6d 2e 76 6e 82  0b 2a 2e 67 6f 6f 67 6c  |.com.vn..*.googl|
    65 2e 64 65 82 0b 2a 2e  67 6f 6f 67 6c 65 2e 65  |e.de..*.google.e|
    73 82 0b 2a 2e 67 6f 6f  67 6c 65 2e 66 72 82 0b  |s..*.google.fr..|
    2a 2e 67 6f 6f 67 6c 65  2e 68 75 82 0b 2a 2e 67  |*.google.hu..*.g|
    6f 6f 67 6c 65 2e 69 74  82 0b 2a 2e 67 6f 6f 67  |oogle.it..*.goog|
    6c 65 2e 6e 6c 82 0b 2a  2e 67 6f 6f 67 6c 65 2e  |le.nl..*.google.|
    70 6c 82 0b 2a 2e 67 6f  6f 67 6c 65 2e 70 74 82  |pl..*.google.pt.|
    12 2a 2e 67 6f 6f 67 6c  65 61 64 61 70 69 73 2e  |.*.googleadapis.|
    63 6f 6d 82 0f 2a 2e 67  6f 6f 67 6c 65 61 70 69  |com..*.googleapi|
    73 2e 63 6e 82 14 2a 2e  67 6f 6f 67 6c 65 63 6f  |s.cn..*.googleco|
    6d 6d 65 72 63 65 2e 63  6f 6d 82 11 2a 2e 67 6f  |mmerce.com..*.go|
    6f 67 6c 65 76 69 64 65  6f 2e 63 6f 6d 82 0c 2a  |oglevideo.com..*|
    2e 67 73 74 61 74 69 63  2e 63 6e 82 0d 2a 2e 67  |.gstatic.cn..*.g|
    73 74 61 74 69 63 2e 63  6f 6d 82 0a 2a 2e 67 76  |static.com..*.gv|
    74 31 2e 63 6f 6d 82 0a  2a 2e 67 76 74 32 2e 63  |t1.com..*.gvt2.c|
    6f 6d 82 14 2a 2e 6d 65  74 72 69 63 2e 67 73 74  |om..*.metric.gst|
    61 74 69 63 2e 63 6f 6d  82 0c 2a 2e 75 72 63 68  |atic.com..*.urch|
    69 6e 2e 63 6f 6d 82 10  2a 2e 75 72 6c 2e 67 6f  |in.com..*.url.go|
    6f 67 6c 65 2e 63 6f 6d  82 16 2a 2e 79 6f 75 74  |ogle.com..*.yout|
    75 62 65 2d 6e 6f 63 6f  6f 6b 69 65 2e 63 6f 6d  |ube-nocookie.com|
    82 0d 2a 2e 79 6f 75 74  75 62 65 2e 63 6f 6d 82  |..*.youtube.com.|
    16 2a 2e 79 6f 75 74 75  62 65 65 64 75 63 61 74  |.*.youtubeeducat|
    69 6f 6e 2e 63 6f 6d 82  07 2a 2e 79 74 2e 62 65  |ion.com..*.yt.be|
    82 0b 2a 2e 79 74 69 6d  67 2e 63 6f 6d 82 1a 61  |..*.ytimg.com..a|
    6e 64 72 6f 69 64 2e 63  6c 69 65 6e 74 73 2e 67  |ndroid.clients.g|
    6f 6f 67 6c 65 2e 63 6f  6d 82 0b 61 6e 64 72 6f  |oogle.com..andro|
    69 64 2e 63 6f 6d 82 1b  64 65 76 65 6c 6f 70 65  |id.com..develope|
    72 2e 61 6e 64 72 6f 69  64 2e 67 6f 6f 67 6c 65  |r.android.google|
    2e 63 6e 82 1c 64 65 76  65 6c 6f 70 65 72 73 2e  |.cn..developers.|
    61 6e 64 72 6f 69 64 2e  67 6f 6f 67 6c 65 2e 63  |android.google.c|
    6e 82 04 67 2e 63 6f 82  06 67 6f 6f 2e 67 6c 82  |n..g.co..goo.gl.|
    14 67 6f 6f 67 6c 65 2d  61 6e 61 6c 79 74 69 63  |.google-analytic|
    73 2e 63 6f 6d 82 0a 67  6f 6f 67 6c 65 2e 63 6f  |s.com..google.co|
    6d 82 12 67 6f 6f 67 6c  65 63 6f 6d 6d 65 72 63  |m..googlecommerc|
    65 2e 63 6f 6d 82 18 73  6f 75 72 63 65 2e 61 6e  |e.com..source.an|
    64 72 6f 69 64 2e 67 6f  6f 67 6c 65 2e 63 6e 82  |droid.google.cn.|
    0a 75 72 63 68 69 6e 2e  63 6f 6d 82 0a 77 77 77  |.urchin.com..www|
    2e 67 6f 6f 2e 67 6c 82  08 79 6f 75 74 75 2e 62  |.goo.gl..youtu.b|
    65 82 0b 79 6f 75 74 75  62 65 2e 63 6f 6d 82 14  |e..youtube.com..|
    79 6f 75 74 75 62 65 65  64 75 63 61 74 69 6f 6e  |youtubeeducation|
    2e 63 6f 6d 82 05 79 74  2e 62 65 30 68 06 08 2b  |.com..yt.be0h..+|
    06 01 05 05 07 01 01 04  5c 30 5a 30 2b 06 08 2b  |........\0Z0+..+|
    06 01 05 05 07 30 02 86  1f 68 74 74 70 3a 2f 2f  |.....0...http://|
    70 6b 69 2e 67 6f 6f 67  6c 65 2e 63 6f 6d 2f 47  |pki.google.com/G|
    49 41 47 32 2e 63 72 74  30 2b 06 08 2b 06 01 05  |IAG2.crt0+..+...|
    05 07 30 01 86 1f 68 74  74 70 3a 2f 2f 63 6c 69  |..0...http://cli|
    65 6e 74 73 31 2e 67 6f  6f 67 6c 65 2e 63 6f 6d  |ents1.google.com|
    2f 6f 63 73 70 30 1d 06  03 55 1d 0e 04 16 04 14  |/ocsp0...U......|
    59 91 47 d7 8b 73 e2 94  2b 52 0a 01 e8 9a 31 d8  |Y.G..s..+R....1.|
    0c 83 6e 24 30 0c 06 03  55 1d 13 01 01 ff 04 02  |..n$0...U.......|
    30 00 30 1f 06 03 55 1d  23 04 18 30 16 80 14 4a  |0.0...U.#..0...J|
    dd 06 16 1b bc f6 68 b5  76 f5 81 b6 bb 62 1a ba  |......h.v....b..|
    5a 81 2f 30 21 06 03 55  1d 20 04 1a 30 18 30 0c  |Z./0!..U....0.0.|
    06 0a 2b 06 01 04 01 d6  79 02 05 01 30 08 06 06  |..+.....y...0...|
    67 81 0c 01 02 02 30 30  06 03 55 1d 1f 04 29 30  |g.....00..U...)0|
    27 30 25 a0 23 a0 21 86  1f 68 74 74 70 3a 2f 2f  |'0%.#.!..http://|
    70 6b 69 2e 67 6f 6f 67  6c 65 2e 63 6f 6d 2f 47  |pki.google.com/G|
    49 41 47 32 2e 63 72 6c  30 0d 06 09 2a 86 48 86  |IAG2.crl0...*.H.|
    f7 0d 01 01 0b 05 00 03  82 01 01 00 49 9c b9 8c  |............I...|
    2d be ea fb 41 da 0a c0  a6 6c 02 d1 5c 2e 74 60  |-...A....l..\.t`|
    56 37 6b e3 69 0e 19 f2  0b 44 08 bd 46 ed f3 99  |V7k.i....D..F...|
    53 56 dc 83 50 6a 81 2c  11 1e 61 e9 c6 fd b2 18  |SV..Pj.,..a.....|
    de 0c 6c 0c 0c 85 d2 fa  a2 66 37 59 3b d4 3f fd  |..l......f7Y;.?.|
    6d 37 db d0 bb 43 1e 40  04 bc ea d8 03 d3 f9 00  |m7...C.@........|
    6d d6 4f ee 30 31 b6 ae  48 c8 f7 9a 81 41 86 9a  |m.O.01..H....A..|
    f3 df c6 0f 04 86 d9 87  ca b8 d6 46 a2 3b b9 c9  |...........F.;..|
    56 8a 88 4d 07 67 9e 61  8f ee 90 eb d8 f8 2d ba  |V..M.g.a......-.|
    3b d7 e2 2d c1 ab b5 7b  f4 14 65 11 4e 68 b9 81  |;..-...{..e.Nh..|
    8d 67 c5 00 b5 33 4c aa  7b d2 1a 1b 7e cb e3 2c  |.g...3L.{...~..,|
    e4 ad 17 36 f5 62 8c ca  6f 7d be aa b8 d4 c2 1b  |...6.b..o}......|
    2d 44 bf d3 02 12 1f 25  75 08 79 0f 18 1b cf 1f  |-D.....%u.y.....|
    5b 4c ce d5 e8 c5 cb f7  38 bc c1 72 bc 89 4c fe  |[L......8..r..L.|
    21 6e b6 c1 88 75 85 15  34 53 bc 2e 68 d1 38 d1  |!n...u..4S..h.8.|
    62 65 d3 47 23 a3 4e 84  7b 0c 9c 8b db 20 fa 7a  |be.G#.N.{......z|
    01 cb 09 7c 0e 99 8a 79  44 8d eb d0 00 04 2c 30  |...|...yD.....,0|
    82 04 28 30 82 03 10 a0  03 02 01 02 02 10 01 00  |..(0............|
    21 25 88 b0 fa 59 a7 77  ef 05 7b 66 27 df 30 0d  |!%...Y.w..{f'.0.|
    06 09 2a 86 48 86 f7 0d  01 01 0b 05 00 30 42 31  |..*.H........0B1|
    0b 30 09 06 03 55 04 06  13 02 55 53 31 16 30 14  |.0...U....US1.0.|
    06 03 55 04 0a 13 0d 47  65 6f 54 72 75 73 74 20  |..U....GeoTrust.|
    49 6e 63 2e 31 1b 30 19  06 03 55 04 03 13 12 47  |Inc.1.0...U....G|
    65 6f 54 72 75 73 74 20  47 6c 6f 62 61 6c 20 43  |eoTrust.Global.C|
    41 30 1e 17 0d 31 37 30  35 32 32 31 31 33 32 33  |A0...17052211323|
    37 5a 17 0d 31 38 31 32  33 31 32 33 35 39 35 39  |7Z..181231235959|
    5a 30 49 31 0b 30 09 06  03 55 04 06 13 02 55 53  |Z0I1.0...U....US|
    31 13 30 11 06 03 55 04  0a 13 0a 47 6f 6f 67 6c  |1.0...U....Googl|
    65 20 49 6e 63 31 25 30  23 06 03 55 04 03 13 1c  |e.Inc1%0#..U....|
    47 6f 6f 67 6c 65 20 49  6e 74 65 72 6e 65 74 20  |Google.Internet.|
    41 75 74 68 6f 72 69 74  79 20 47 32 30 82 01 22  |Authority.G20.."|
    30 0d 06 09 2a 86 48 86  f7 0d 01 01 01 05 00 03  |0...*.H.........|
    82 01 0f 00 30 82 01 0a  02 82 01 01 00 9c 2a 04  |....0.........*.|
    77 5c d8 50 91 3a 06 a3  82 e0 d8 50 48 bc 89 3f  |w\.P.:.....PH..?|
    f1 19 70 1a 88 46 7e e0  8f c5 f1 89 ce 21 ee 5a  |..p..F~......!.Z|
    fe 61 0d b7 32 44 89 a0  74 0b 53 4f 55 a4 ce 82  |.a..2D..t.SOU...|
    62 95 ee eb 59 5f c6 e1  05 80 12 c4 5e 94 3f bc  |b...Y_......^.?.|
    5b 48 38 f4 53 f7 24 e6  fb 91 e9 15 c4 cf f4 53  |[H8.S.$........S|
    0d f4 4a fc 9f 54 de 7d  be a0 6b 6f 87 c0 d0 50  |..J..T.}..ko...P|
    1f 28 30 03 40 da 08 73  51 6c 7f ff 3a 3c a7 37  |.(0.@..sQl..:<.7|
    06 8e bd 4b 11 04 eb 7d  24 de e6 f9 fc 31 71 fb  |...K...}$....1q.|
    94 d5 60 f3 2e 4a af 42  d2 cb ea c4 6a 1a b2 cc  |..`..J.B....j...|
    53 dd 15 4b 8b 1f c8 19  61 1f cd 9d a8 3e 63 2b  |S..K....a....>c+|
    84 35 69 65 84 c8 19 c5  46 22 f8 53 95 be e3 80  |.5ie....F".S....|
    4a 10 c6 2a ec ba 97 20  11 c7 39 99 10 04 a0 f0  |J..*......9.....|
    61 7a 95 25 8c 4e 52 75  e2 b6 ed 08 ca 14 fc ce  |az.%.NRu........|
    22 6a b3 4e cf 46 03 97  97 03 7e c0 b1 de 7b af  |"j.N.F....~...{.|
    45 33 cf ba 3e 71 b7 de  f4 25 25 c2 0d 35 89 9d  |E3..>q...%%..5..|
    9d fb 0e 11 79 89 1e 37  c5 af 8e 72 69 02 03 01  |....y..7...ri...|
    00 01 a3 82 01 11 30 82  01 0d 30 1f 06 03 55 1d  |......0...0...U.|
    23 04 18 30 16 80 14 c0  7a 98 68 8d 89 fb ab 05  |#..0....z.h.....|
    64 0c 11 7d aa 7d 65 b8  ca cc 4e 30 1d 06 03 55  |d..}.}e...N0...U|
    1d 0e 04 16 04 14 4a dd  06 16 1b bc f6 68 b5 76  |......J......h.v|
    f5 81 b6 bb 62 1a ba 5a  81 2f 30 0e 06 03 55 1d  |....b..Z./0...U.|
    0f 01 01 ff 04 04 03 02  01 06 30 2e 06 08 2b 06  |..........0...+.|
    01 05 05 07 01 01 04 22  30 20 30 1e 06 08 2b 06  |......."0.0...+.|
    01 05 05 07 30 01 86 12  68 74 74 70 3a 2f 2f 67  |....0...http://g|
    2e 73 79 6d 63 64 2e 63  6f 6d 30 12 06 03 55 1d  |.symcd.com0...U.|
    13 01 01 ff 04 08 30 06  01 01 ff 02 01 00 30 35  |......0.......05|
    06 03 55 1d 1f 04 2e 30  2c 30 2a a0 28 a0 26 86  |..U....0,0*.(.&.|
    24 68 74 74 70 3a 2f 2f  67 2e 73 79 6d 63 62 2e  |$http://g.symcb.|
    63 6f 6d 2f 63 72 6c 73  2f 67 74 67 6c 6f 62 61  |com/crls/gtgloba|
    6c 2e 63 72 6c 30 21 06  03 55 1d 20 04 1a 30 18  |l.crl0!..U....0.|
    30 0c 06 0a 2b 06 01 04  01 d6 79 02 05 01 30 08  |0...+.....y...0.|
    06 06 67 81 0c 01 02 02  30 1d 06 03 55 1d 25 04  |..g.....0...U.%.|
    16 30 14 06 08 2b 06 01  05 05 07 03 01 06 08 2b  |.0...+.........+|
    06 01 05 05 07 03 02 30  0d 06 09 2a 86 48 86 f7  |.......0...*.H..|
    0d 01 01 0b 05 00 03 82  01 01 00 ca 49 e5 ac d7  |............I...|
    64 64 77 5b be 71 fa cf  f4 1e 23 c7 9a 69 63 54  |ddw[.q....#..icT|
    5f eb 4c d6 19 28 23 64  66 8e 1c c7 87 80 64 5f  |_.L..(#df.....d_|
    04 8b 26 af 98 df 0a 70  bc bc 19 3d ee 7b 33 a9  |..&....p...=.{3.|
    7f bd f4 05 d4 70 bb 05  26 79 ea 9a c7 98 b9 07  |.....p..&y......|
    19 65 34 cc 3c e9 3f c5  01 fa 6f 0c 7e db 7a 70  |.e4.<.?...o.~.zp|
    5c 4c fe 2d 00 f0 ca be  2d 8e b4 a8 80 fb 01 13  |\L.-....-.......|
    88 cb 9c 3f e5 bb 77 ca  3a 67 36 f3 ce d5 27 02  |...?..w.:g6...'.|
    72 43 a0 bd 6e 02 f1 47  05 71 3e 01 59 e9 11 9e  |rC..n..G.q>.Y...|
    1a f3 84 0f 80 a6 a2 78  35 2f b6 c7 a2 7f 17 7c  |.......x5/.....||
    e1 8b 56 ae ee 67 88 51  27 30 60 a5 62 52 c3 37  |..V..g.Q'0`.bR.7|
    d5 3b ea 85 2a 01 38 87  a2 cf 70 ad a4 7a c9 c4  |.;..*.8...p..z..|
    e7 ca c5 da bc 23 32 f2  fe 18 c2 7b e0 df 3b 2f  |.....#2....{..;/|
    d4 d0 10 e6 96 4c fb 44  b7 21 64 0d b9 00 94 30  |.....L.D.!d....0|
    12 26 87 58 98 39 05 38  0f cc 82 48 0c 0a 47 66  |.&.X.9.8...H..Gf|
    ee bf b4 5f c4 ff 70 a8  e1 7f 8b 79 2b b8 65 32  |..._..p....y+.e2|
    a3 b9 b7 31 e9 0a f5 f6  1f 32 dc 00 03 81 30 82  |...1.....2....0.|
    03 7d 30 82 02 e6 a0 03  02 01 02 02 03 12 bb e6  |.}0.............|
    30 0d 06 09 2a 86 48 86  f7 0d 01 01 05 05 00 30  |0...*.H........0|
    4e 31 0b 30 09 06 03 55  04 06 13 02 55 53 31 10  |N1.0...U....US1.|
    30 0e 06 03 55 04 0a 13  07 45 71 75 69 66 61 78  |0...U....Equifax|
    31 2d 30 2b 06 03 55 04  0b 13 24 45 71 75 69 66  |1-0+..U...$Equif|
    61 78 20 53 65 63 75 72  65 20 43 65 72 74 69 66  |ax.Secure.Certif|
    69 63 61 74 65 20 41 75  74 68 6f 72 69 74 79 30  |icate.Authority0|
    1e 17 0d 30 32 30 35 32  31 30 34 30 30 30 30 5a  |...020521040000Z|
    17 0d 31 38 30 38 32 31  30 34 30 30 30 30 5a 30  |..180821040000Z0|
    42 31 0b 30 09 06 03 55  04 06 13 02 55 53 31 16  |B1.0...U....US1.|
    30 14 06 03 55 04 0a 13  0d 47 65 6f 54 72 75 73  |0...U....GeoTrus|
    74 20 49 6e 63 2e 31 1b  30 19 06 03 55 04 03 13  |t.Inc.1.0...U...|
    12 47 65 6f 54 72 75 73  74 20 47 6c 6f 62 61 6c  |.GeoTrust.Global|
    20 43 41 30 82 01 22 30  0d 06 09 2a 86 48 86 f7  |.CA0.."0...*.H..|
    0d 01 01 01 05 00 03 82  01 0f 00 30 82 01 0a 02  |...........0....|
    82 01 01 00 da cc 18 63  30 fd f4 17 23 1a 56 7e  |.......c0...#.V~|
    5b df 3c 6c 38 e4 71 b7  78 91 d4 bc a1 d8 4c f8  |[.<l8.q.x.....L.|
    a8 43 b6 03 e9 4d 21 07  08 88 da 58 2f 66 39 29  |.C...M!....X/f9)|
    bd 05 78 8b 9d 38 e8 05  b7 6a 7e 71 a4 e6 c4 60  |..x..8...j~q...`|
    a6 b0 ef 80 e4 89 28 0f  9e 25 d6 ed 83 f3 ad a6  |......(..%......|
    91 c7 98 c9 42 18 35 14  9d ad 98 46 92 2e 4f ca  |....B.5....F..O.|
    f1 87 43 c1 16 95 57 2d  50 ef 89 2d 80 7a 57 ad  |..C...W-P..-.zW.|
    f2 ee 5f 6b d2 00 8d b9  14 f8 14 15 35 d9 c0 46  |.._k........5..F|
    a3 7b 72 c8 91 bf c9 55  2b cd d0 97 3e 9c 26 64  |.{r....U+...>.&d|
    cc df ce 83 19 71 ca 4e  e6 d4 d5 7b a9 19 cd 55  |.....q.N...{...U|
    de c8 ec d2 5e 38 53 e5  5c 4f 8c 2d fe 50 23 36  |....^8S.\O.-.P#6|
    fc 66 e6 cb 8e a4 39 19  00 b7 95 02 39 91 0b 0e  |.f....9.....9...|
    fe 38 2e d1 1d 05 9a f6  4d 3e 6f 0f 07 1d af 2c  |.8......M>o....,|
    1e 8f 60 39 e2 fa 36 53  13 39 d4 5e 26 2b db 3d  |..`9..6S.9.^&+.=|
    a8 14 bd 32 eb 18 03 28  52 04 71 e5 ab 33 3d e1  |...2...(R.q..3=.|
    38 bb 07 36 84 62 9c 79  ea 16 30 f4 5f c0 2b e8  |8..6.b.y..0._.+.|
    71 6b e4 f9 02 03 01 00  01 a3 81 f0 30 81 ed 30  |qk..........0..0|
    1f 06 03 55 1d 23 04 18  30 16 80 14 48 e6 68 f9  |...U.#..0...H.h.|
    2b d2 b2 95 d7 47 d8 23  20 10 4f 33 98 90 9f d4  |+....G.#..O3....|
    30 1d 06 03 55 1d 0e 04  16 04 14 c0 7a 98 68 8d  |0...U.......z.h.|
    89 fb ab 05 64 0c 11 7d  aa 7d 65 b8 ca cc 4e 30  |....d..}.}e...N0|
    0f 06 03 55 1d 13 01 01  ff 04 05 30 03 01 01 ff  |...U.......0....|
    30 0e 06 03 55 1d 0f 01  01 ff 04 04 03 02 01 06  |0...U...........|
    30 3a 06 03 55 1d 1f 04  33 30 31 30 2f a0 2d a0  |0:..U...3010/.-.|
    2b 86 29 68 74 74 70 3a  2f 2f 63 72 6c 2e 67 65  |+.)http://crl.ge|
    6f 74 72 75 73 74 2e 63  6f 6d 2f 63 72 6c 73 2f  |otrust.com/crls/|
    73 65 63 75 72 65 63 61  2e 63 72 6c 30 4e 06 03  |secureca.crl0N..|
    55 1d 20 04 47 30 45 30  43 06 04 55 1d 20 00 30  |U...G0E0C..U...0|
    3b 30 39 06 08 2b 06 01  05 05 07 02 01 16 2d 68  |;09..+........-h|
    74 74 70 73 3a 2f 2f 77  77 77 2e 67 65 6f 74 72  |ttps://www.geotr|
    75 73 74 2e 63 6f 6d 2f  72 65 73 6f 75 72 63 65  |ust.com/resource|
    73 2f 72 65 70 6f 73 69  74 6f 72 79 30 0d 06 09  |s/repository0...|
    2a 86 48 86 f7 0d 01 01  05 05 00 03 81 81 00 76  |*.H............v|
    e1 12 6e 4e 4b 16 12 86  30 06 b2 81 08 cf f0 08  |..nNK...0.......|
    c7 c7 71 7e 66 ee c2 ed  d4 3b 1f ff f0 f0 c8 4e  |..q~f....;.....N|
    d6 43 38 b0 b9 30 7d 18  d0 55 83 a2 6a cb 36 11  |.C8..0}..U..j.6.|
    9c e8 48 66 a3 6d 7f b8  13 d4 47 fe 8b 5a 5c 73  |..Hf.m....G..Z\s|
    fc ae d9 1b 32 19 38 ab  97 34 14 aa 96 d2 eb a3  |....2.8..4......|
    1c 14 08 49 b6 bb e5 91  ef 83 36 eb 1d 56 6f ca  |...I......6..Vo.|
    da bc 73 63 90 e4 7f 7b  3e 22 cb 3d 07 ed 5f 38  |..sc...{>".=.._8|
    74 9c e3 03 50 4e a1 af  98 ee 61 f2 84 3f 12     |t...PN....a..?. |

```

**Comment by bassosimone** [comment]
[Measurement-kit/mkcurl](https://github.com/measurement-kit/mkcurl), included in MK v0.9.0-alpha.10, gives you the full certificate chain, via cURL. When WebConnectivity will be switched over to using `mkcurl` as backend, we will have this functionality available.

**Comment by bassosimone** [comment]
Superseded by https://github.com/ooni/probe-engine/issues/17
